sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -219,7 +223,8 @@ int hda_dsp_ctrl_init_chip(struct snd_sof_dev *sdev, bool full_reset)
 	/* check to see if controller is ready */
 	if (!snd_hdac_chip_readb(bus, GCTL)) {
 		dev_dbg(bus->dev, "controller not ready!\n");
-		return -EBUSY;
+		ret = -EBUSY;
+		goto err;
```

Comment:
this changes the sequence now, in the err: handling you will do a 	hda_dsp_ctrl_misc_clock_gating(sdev, true); which wasn't done before. Or the err: label location is off by one line?

I would suggest splitting the error flow corrections from the chip_init stuff, that's confusing here, no?

Reply:
@plbossart I think it was a miss in the current code to not set clock_gating to true in case of error no?

Reply:
@ranj063 ack indeed. The existing code leaves the misc clock gating off if init fails. When I added the error case for the codec wakeline, it looked a bit wrong to unroll only part of the change. But ack, let me check the impacts again and submit as separate patch to make this more clear. The codec wakeline we have to unroll for sure as that has impact to other IP. The misc clock control is less obvious as audio IP is left anyways into uninitialized state when this function fails.

Reply:
@plbossart @ranj063 now split into a separate patch

Reply:
ok makes sense then!